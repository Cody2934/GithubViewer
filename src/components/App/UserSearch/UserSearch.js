import React from 'react';
import PropTypes from 'prop-types';

const UserSearch = ({ username, onUsernameChange }) => (
  <>
    <input type="username" value={username} onChange={onUsernameChange} />
  </>
);

UserSearch.propTypes = {
  username: PropTypes.string.isRequired,
  onUsernameChange: PropTypes.func.isRequired
};

export default UserSearch;
