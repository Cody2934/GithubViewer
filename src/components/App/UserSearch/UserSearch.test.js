import React from 'react';
import { shallow } from 'enzyme';
import UserSearch from './UserSearch';

describe('UserSearch', () => {
  it('matches a snapshot', () => {
    const wrapper = shallow(<UserSearch 
      username="Cody"
      onUsernameChange={() => {}}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
