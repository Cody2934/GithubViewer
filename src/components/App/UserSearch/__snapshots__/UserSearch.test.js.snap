// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserSearch matches a snapshot 1`] = `
<Fragment>
  <input
    onChange={[Function]}
    type="username"
    value="Cody"
  />
</Fragment>
`;
